lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,49 @@
+Facter.add(:powershell_installed) do
+ confine :operatingsystem => :windows
+ require 'win32/registry'
+ setcode do
+  begin
+    Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\PowerShell\1\PowerShellEngine', ::Win32::Registry::KEY_READ)
```

Comment:
This key will be in both x64 and Wow6432Node, so I don't know if we need to turn off registry redirection. Prolly for safety we should.